mlir/lib/Dialect/OpenACC/IR/OpenACC.cpp

```diff
@@ -389,6 +389,87 @@ void OpenACCDialect::initialize() {
   memref::GlobalOp::attachInterface<MemrefGlobalVariableModel>(*getContext());
 }
 
+//===----------------------------------------------------------------------===//
+// RegionBranchOpInterface for acc.kernels / acc.parallel / acc.serial /
+// acc.kernel_environment / acc.data / acc.host_data / acc.loop
+//===----------------------------------------------------------------------===//
+
+void KernelsOp::getSuccessorRegions(RegionBranchPoint point,
+                                    SmallVectorImpl<RegionSuccessor> &regions) {
+  if (point.isParent()) {
+    regions.push_back(RegionSuccessor(&getRegion()));
+    return;
+  }
+
+  regions.push_back(
+      RegionSuccessor(getOperation(), getOperation()->getResults()));
```

Comment:
I suggest extracting this into a static templated utility function and just calling it in all methods, e.g.
```
template <typename OP>
static void getSuccessorRegionsForContainerOp(Op op, RegionBranchPoint, SmallVectorImpl<RegionSucessor> &);
```